formula/src/main/java/com/instacart/formula/internal/FormulaManager.kt

```diff
@@ -13,34 +13,24 @@ interface FormulaManager<Input, Output> {
     /**
      * Creates the current [Output] and prepares the next frame that will need to be processed.
      */
-    fun evaluate(
-        input: Input,
-        transitionId: TransitionId
-    ): Evaluation<Output>
-
-    /**
-     * Method updates the transition id of the existing evaluation. This
-     * method is called when there were no changes and evaluation does
-     * not need to run in this part of the formula tree.
-     */
-    fun updateTransitionId(transitionId: TransitionId)
+    fun evaluate(input: Input): Evaluation<Output>
```

Comment:
Removing global `TransitionId` allows us to simplify quite a few functions and also allows us to remove `updateTransitionId` which would have to iterate across the whole formula tree.